### PR TITLE
Created fine-grained FA definitions

### PR DESCRIPTION
**Pull request by ethouris**

### CODE REVIEW

File: apps/logsupport_appdefs.cpp [hunk 1]
```diff
@@ -12,12 +12,39 @@
...
-    Install("DATA", SRT_LOGFA_DATA);
+    Install("SOCKMGMT", SRT_LOGFA_SOCKMGMT);
+    Install("CONN", SRT_LOGFA_CONN);
+    Install("XTIMER", SRT_LOGFA_XTIMER);
     Install("TSBPD", SRT_LOGFA_TSBPD);
-    Install("REXMIT", SRT_LOGFA_REXMIT);
+    Install("RSRC", SRT_LOGFA_RSRC);
 
     Install("CONGEST", SRT_LOGFA_CONGEST);
+    Install("PFILTER", SRT_LOGFA_PFILTER);
```
Comment (maxsharabayko) [inline apps/logsupport_appdefs.cpp:22]: Might be helpful to have somedescription guidline and decoding of abbreviations in comments.

File: srtcore/srt.h [hunk 2]
```diff
@@ -571,22 +571,69 @@ enum SRT_REJECT_REASON
...
 
-#define SRT_LOGFA_CONTROL    2 // mglog
-#define SRT_LOGFA_DATA       3 // dlog
+#define SRT_LOGFA_GENERAL    0 // gglog
+#define SRT_LOGFA_SOCKMGMT   1 // smlog
+#define SRT_LOGFA_CONN       2 // calog
+#define SRT_LOGFA_XTIMER     3 // xtlog
 #define SRT_LOGFA_TSBPD      4 // tslog
-#define SRT_LOGFA_REXMIT     5 // rxlog
+#define SRT_LOGFA_RSRC       5 // rslog
```
Comment (maxsharabayko) [inline srtcore/srt.h:600]: And here as well some guide on which FA corresponds to which part of the library.

File: apps/logsupport_appdefs.cpp (hunk 1 above)
Comment (ethouris) [reply apps/logsupport_appdefs.cpp:22]: Yes, but I think not in generated files.

I was thinking about putting the sole definition of the FA in a separate file, just read by the generator script. What do you think?

File: srtcore/srt.h (hunk 2 above)
Comment (ethouris) [reply srtcore/srt.h:600]: Description might be a part of the FA definition... :thinking:

**Review by maxsharabayko — COMMENTED** [review]
Looks good except for naming. Not sure if it is worth renaming as it requires additional time may introduce issues.
For example, these names are a bit misleading:
- iplog: Internal activities not connected directly to a socket
- calog: Connection establishment and handshake -> `cnlog`?

Maybe there is no need in the `log` suffix, and all five chars could be used instead.
E.g. CChannel: receiving side can be renamed from `krlog` to `rchan` and `schan` for sending side.
But up to you.

**Pull request merged**